controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -256,15 +256,29 @@ func (a *Adapter) determineIfAllIntegrationPipelinesPassed(integrationPipelineRu
 	allIntegrationPipelineRunsPassed := true
 	for _, integrationPipelineRun := range *integrationPipelineRuns {
 		integrationPipelineRun := integrationPipelineRun // G601
-		pipelineRunOutcome, err := h.CalculateIntegrationPipelineRunOutcome(a.client, a.context, a.logger.Logger, &integrationPipelineRun)
+		if !h.HasPipelineRunFinished(&integrationPipelineRun) {
+			a.logger.Info("The integration PipelineRun has not finished yet")
+			allIntegrationPipelineRunsPassed = false
+			continue
+		}
+		pipelineRunOutcome, err := h.GetIntegrationPipelineRunOutcome(a.client, a.context, &integrationPipelineRun)
 		if err != nil {
 			a.logger.Error(err, "Failed to get Integration PipelineRun outcome")
 			return false, err
 		}
-		if !pipelineRunOutcome {
-			a.logger.Info("Integration PipelineRun did not pass all tests")
+		if !pipelineRunOutcome.GetOutcome() {
+			if !pipelineRunOutcome.HasPipelineRunSucceeded() {
+				a.logger.Error(
+					fmt.Errorf("integration PipelineRun failed: %s", h.GetPipelineRunFailedReason(&integrationPipelineRun)),
+					"PipelineRun failed without test results of TaskRuns",
+				)
+
+			} else {
+				a.logger.Info("Integration PipelineRun did not pass all tests")
```

Comment:
I've to change all logs back to explicitly log pipeline name, inherited pipeline name in logs keys is the one being reconciled. It's puzzling